apps/chat/src/components/Marketplace/Marketplace.tsx

```diff
@@ -31,10 +38,19 @@ export const Marketplace = () => {
   );
   const isProfileOpen = useAppSelector(UISelectors.selectIsProfileOpen);
   const isModelsLoading = useAppSelector(ModelsSelectors.selectModelsIsLoading);
+  const applyModelStatus = useAppSelector(
+    MarketplaceSelectors.selectApplyModelStatus,
+  );
+
   const [isMobile, setIsMobile] = useState(isSmallScreen());
 
   const showOverlay = (isFilterbarOpen || isProfileOpen) && isSmallScreen();
 
+  const isLoading =
+    isModelsLoading ||
+    (applyModelStatus !== CompletionStatus.PENDING &&
```

Comment:
I suppose it's typo?
```suggestion
    (applyModelStatus !== CompletionStatus.COMPLETED &&
```

Reply:
Not a typo, loader shown when process is started, PENDING means process is not started yet. Reason - COMPLEATED is set after model is applied, but there is also redirect that takes time, so we need to wait for redirect to set completion status to default which is PENDING, if we hide loader on COMPLEATED loader will be hidden before redirect is finished